### PR TITLE
Adding an option for providing user theming

### PR DESCRIPTION
**Pull request by alexagranov**
Just nice to be able to provide my own logo, etc.

### CODE REVIEW

File: app/views/grape_swagger_rails/application/index.html.erb [hunk 1]
```diff
@@ -4,6 +4,9 @@
   <title><%= GrapeSwaggerRails.options.app_name || 'Swagger UI' %></title>
   <link href='//fonts.googleapis.com/css?family=Droid+Sans:400,700' rel='stylesheet' type='text/css'/>
   <%= stylesheet_link_tag 'grape_swagger_rails/application.css' %>
+  <% if GrapeSwaggerRails.options[:custom_css_file] %>
```
Comment (unloved) [inline app/views/grape_swagger_rails/application/index.html.erb]: seems it's gonna be true with default '' value

**Comment by dblock** [comment]
This would need tests and a CHANGELOG entry to be merged (we're trying to be more disciplined here ;)).

That said, it seems hacky. I wonder whether we'd be better served to refactoring the ERB into partials and to let you assemble a custom index.html.erb yourself out of them. You don't have to use this ERB already, you might as well roll your own that looks the way you want, but we could make it easier to do.

**Comment by alexagranov** [comment]
Absolutely.  It was late and I needed to keep moving.  Consider it a
feature request rather than a contribution. ;-)

On Fri, Feb 6, 2015 at 5:44 PM, Daniel Doubrovkine (dB.) @dblockdotorg <
notifications@github.com> wrote:

> This would need tests and a CHANGELOG entry to be merged (we're trying to
> be more disciplined here ;)).
>
> That said, it seems hacky. I wonder whether we'd be better served to
> refactoring the ERB into partials and to let you assemble a custom
> index.html.erb yourself out of them. You don't have to use this ERB
> already, you might as well roll your own that looks the way you want, but
> we could make it easier to do.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/TinkerDev/grape-swagger-rails/pull/3#issuecomment-73327029>
> .
>



-- 
*Alex Agranov*
Director/Sr.Engineer
Morphogenic Productions
www.morphogenic.net
www.linkedin.com/in/alexagranov
(917) 509-1856

File: app/views/grape_swagger_rails/application/index.html.erb (hunk 1 above)
Comment (alexagranov) [inline app/views/grape_swagger_rails/application/index.html.erb]: fixed with commit ca7f11659400b157735ee52e737b37ec5b0879e5

**Comment by alexagranov** [comment]
In case you still want to merge, cleaning up my late night sloppiness.

**Comment by dblock** [comment]
Thanks @alexagranov! I think I'd rather not merge this in favor of a more modular approach as described above. Curious to hear what others think, will let this sit here for a bit.

**Comment by pablocrivella** [comment]
+1000!

**Comment by alexagranov** [comment]
Cleaning this up, just realized I've been committing willy nilly...

**Comment by dblock** [comment]
@alexagranov Please do consider contributing a better implementation, this would be a very useful feature!

**Pull request closed without merging**